app/src/main/java/org/greenstand/android/TreeTracker/permissions/Permissions.kt

```diff
@@ -0,0 +1,211 @@
+                                    )
+                                }
+                            }
+                        )
+                    }
+                    else -> {
+                             PermissionDeniedPermanentlyDialog(navController)
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_FINE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        if(state.isLocationEnabled == false){
+                            enableLocation()
+                        }
+                        return
+                    }
+                    perm.shouldShowRationale -> {
+                        LocationRationaleDialog(navController = navController, perm = perm)
+                    }
+                    else -> {
+                            PermissionDeniedPermanentlyDialog(navController)
+                    }
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        if(viewModel.isLocationEnabled() == false){
+                            enableLocation()
+                        }
+                        return
+                    }
+                    perm.shouldShowRationale -> {
+                        LocationRationaleDialog(navController = navController, perm = perm)
+                    }
+                    else -> {
+                            PermissionDeniedPermanentlyDialog(navController)
+                    }
+                }
+            }
+        }
+    }
+}
+
+
+@ExperimentalPermissionsApi
+@Composable
+fun LocationRationaleDialog(navController: NavHostController, perm: PermissionState) {
+    AlertDialog(
+        onDismissRequest = { navController.popBackStack() },
+        title = {
+            Text(text = stringResource(R.string.accept_location_permission_header))
+        },
+        text = {
+            Text(
+                text = stringResource(R.string.accept_location_permission_message),
+                color = Color.Green
+            )
+        },
+        buttons = {
+            Button(
+                modifier = Modifier.wrapContentSize(),
+                onClick = {
+                    perm.launchPermissionRequest()
+                    navController.popBackStack()
+                }
+            ) {
+                Text(
+                    text = stringResource(R.string.accept_permission)
+                )
+            }
+        }
+    )
+}
+
+@Composable
+fun enableLocation() {
+    val activity: Context = LocalContext.current as Activity
+    val intent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+    startActivity(activity, intent, null)
+}
+
+@Composable
+fun PermissionDeniedPermanentlyDialog(navController: NavHostController,viewModel: PermissionViewModel = viewModel(factory = LocalViewModelFactory.current)) {

```

Comment:
This viewmodel is never used

Reply:
The portion of code using it previously has been deleted. I would delete this as well.